src/oscar/templates/oscar/dashboard/ranges/range_product_list.html

```diff
@@ -30,114 +30,248 @@
+          {% endif %}
+
+        </div>
+
+      {% endif %}
 
     </div>
-        {% endif %}
-        <div class="form-actions">
-            <a href="{% url 'dashboard:range-update' pk=range.id %}" class="btn btn-primary">{% trans "Edit range" %}</a> {% trans "or" %}
-            <a href="{% url 'dashboard:range-list' %}" class="">{% trans "return to range list" %}</a>
+
+    <div class="tab-pane fade {% if upload_type == "excluded" %}show active{% endif %}" id="excluded" role="tabpanel">
+
+        <div class="table-header">
+          <h3>{% trans "Exclude products" %}</h3>
         </div>
 
+        <div class="card card-body bg-light">
+
+          <form method="post" class="form-stacked" enctype="multipart/form-data">
+            {% csrf_token %}
+            <input type="hidden" name="action" value="add_excluded_products"/>
+            {% include 'oscar/dashboard/partials/form_fields.html' with form=form_excluded %}
+            <div class="controls">
+              <button type="submit" class="btn btn-primary" data-loading-text="{% trans 'Running...' %}">
+                {% trans "Go!" %}
+              </button>
+            </div>
+          </form>
+
+          {% with uploads=file_uploads_excluded %}
+          {% if uploads %}
+            <div class="table-header">
+              <h3>{% trans "Upload history" %}</h3>
+            </div>
+            <table class="table table-striped table-bordered table-hover">
+              <thead>
+                <tr>
+                  <th>{% trans "Filename" %}</th>
+                  <th>{% trans "New products" %}</th>
+                  <th>{% trans "Duplicate products" %}</th>
+                  <th>{% trans "Unknown products" %}</th>
+                  <th>{% trans "Date uploaded" %}</th>
+                </tr>
+              </thead>
+              <tbody>
+                {% for upload in uploads %}
+                  <tr>
+                    <td>{{ upload.filepath }}</td>
+                    <td>{{ upload.num_new_skus }}</td>
+                    <td>{{ upload.num_duplicate_skus }}</td>
+                    <td>{{ upload.num_unknown_skus }}</td>
+                    <td>{{ upload.date_uploaded }}</td>
+                  </tr>
+                {% endfor %}
+              </tbody>
+            </table>
+          {% endif %}
+          {% endwith %}
+
+          {% if range.excluded_products.count %}
+            <form method="post">
+              {% csrf_token %}
+              <table class="table table-striped table-bordered table-hover">
+               <caption>
+                  <h3 class="float-left">{% trans "Products excluded from range" %}</h3>
+                  <div class="float-right">
+                    <input type="hidden" name="action" value="remove_excluded_products" />
+                    <button type="submit" class="btn btn-secondary" data-loading-text="{% trans 'Removing...' %}">
+                      {% trans "Remove excluded products" %}
```

Comment:
Please change this to:

Remove selected excluded products

Reply:
fixed